lib/Doctrine/ORM/Tools/Pagination/Paginator.php

```diff
@@ -107,6 +107,16 @@ public function count()
 
         return $this->count;
     }
+    
+    /**
+     * Returns last page number
+     *
+     * @return int
+     */
+    public function lastPage()
+    {
+        return ceil($this->count() / $this->query->getMaxResults());
```

Comment:
Since `$this->query->getMaxResults()` is `int|null`, it's prone to division by zero when the value is  `null` or `0`. Needs to be checked explicitly.